MetaMask/metamask-extension
# fix: Blockaid send `External Link Clicked` metric for missing tx types

## **Description**

As [mentioned in the related issue ](https://app.zenhub.com/workspaces/-confirmations-secure-ux-6245e6e2348677001213b8d2/issues/gh/metamask/metamask-planning/1756#issuecomment-1894363798)

"
not all instances of the BlockaidBannerAlert are firing the event.

**instances supported:**
- ui/components/app/signature-request/signature-request.js (eth_signTypedData versions V3 and V4)
- ui/components/app/transaction-alerts/transaction-alerts.js

**instances not supported:**
- ui/components/app/signature-request-original/signature-request-original.component.js
- ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.js
- ui/pages/token-allowance/token-allowance.js
- ui/components/app/signature-request-siwe/signature-request-siwe.js

"

This PR adds support to the missing instances.

Additionally:
- minor lint fixes - missing deps
- remove superfluous ONLY_INCLUDE_IF(blockaid) guardrails


Updates to the schema are being discussed. I am considering cleanup. These and tests will be implemented in a future PR. 

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/22667

## **Manual testing steps**

1. Go to test-dapp
2. Use PPOM section to QA
- confirm approve - Approval BUSD
- confirm allowance - Set Approval For All 

I was unable to reproduce the malicious SIWE and the non-v3 or non-v4 signTyped signatures

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
